doc/savegame.rst

```diff
@@ -27,25 +27,42 @@ Introduction
 ============
 This file documents the file format used by FreeRCT for saving games.
 
+.. note:: The file format changed significantly between the file header versions 9 and 10.
+          This document covers only the format from header version 10 onwards.
+          Documentation for older format versions can be found in the file "savegame_history.rst".
+
 The file starts with a file header to identify the file as being a save game.
-After the file header come data blocks of game elements that are stored.
+After the file header come data patterns of game elements that are stored.
+
+All data in the file is organized in **patterns**. Each pattern starts by
+specifying a version number followed by the pattern's data layout as
+detailed below. A pattern can contain other nested patterns. A pattern
+which is not nested in any other pattern is called a **block**.
 
 ======  ======  =======  ======================================================
 Offset  Length  Version  Description
 ======  ======  =======  ======================================================
-   0      12      1-     File header
-  12      16      1-     Current date block
-  28       ?      3-     Current basic world block.
-   ?      16      1-     Current random number block
-   ?       ?      2-     Current financial data.
-   ?      28      4-     Current weather block.
+   0       ?     10-     File header
+   ?       ?     10-     Current date block.
+   ?       ?     10-     Current basic world block.
+   ?       ?     10-     Current financial data.
+   ?       ?     10-     Current weather block.
+   ?       ?     10-     Rides block.
+   ?       ?     10-     Scenery block.
+   ?       ?     10-     Guests block.
+   ?       ?     10-     Staff block.
+   ?       ?     10-     Inbox block.
+   ?       ?     10-     Current random number block.
    ?                     Total length of the save file.
```

Comment:
Almost everywhere, the total size isn't fixed, so maybe just drop the "total size" line at all?

Arguably the offset column could be deleted too, but that may be useful enough to keep.

Would it be useful to distinguish fields from patterns? Eg make the patterns bold, so a reader knows it's defined else where. Ideally it would be a link, but that may be too much work in REST format.